### ISSUE TITLE
std.mem.Allocator.realloc: broken LLVM module found: call parameter type does not match function signature

### ISSUE THREAD

**Issue by speps** [ziglang/zig#3898]
I encountered an error while doing the Advent of Code 2019.

```
broken LLVM module found: Call parameter type does not match function signature!
  %16 = load %"[]i64"*, %"[]i64"** %mem, align 8, !dbg !15076
 { %"[]i64", i16 }*  call fastcc void @std.mem.Allocator.realloc.174(%"[]i64"* sret %16, %std.builtin.StackTrace* %error_return_trace, %std.mem.Allocator* %17, %"[]i64"* %18, i64 %23), !dbg !15086

This is a bug in the Zig compiler.
Unable to dump stack trace: debug info stripped
```

The simplest repro case I could do is below. I'm new to the language so I don't know if there should be an error reported for this. The goal here is to be able to update the `mem` variable with the result of the `realloc` inside a function.

```zig
const std = @import("std");

var allocator: *std.mem.Allocator = undefined;

fn mem_address(mem: *[]i64, i: usize) void {
    if (mem.*.len < (i + 1)) {
        mem.* = allocator.realloc(mem.*, i + 1);
    }
}

pub fn main() !void {
    var arena = std.heap.ArenaAllocator.init(std.heap.direct_allocator);
    defer arena.deinit();

    allocator = &arena.allocator;

    var mem: []i64 = try allocator.alloc(i64, 4);
    _ = mem_address(&mem, 5);
}
```

**Comment by deingithub** [comment]
Your issue here is likely caused by `allocator.realloc` being a function that can fail and return an error: https://github.com/ziglang/zig/blob/master/lib/std/mem.zig#L40
Try handling the error, for example by suffixing the method invocation with `catch unreachable` to assert that `realloc` will not fail.

This turning up as a rather cryptic LLVM error instead of a failure during analysis is probably still a bug though.

**Comment by speps** [comment]
Indeed thanks for the workaround.

It's happening on 81f1f72197113a45e827d5c984e219a28aa28083 (latest master at time of writing) by the way.

**Comment by pixelherodev** [comment]
This isn't a bug with Zig, not really.

This *is* explicitly a bug with your code - that error *must* be caught. The error here is that Zig isn't properly reporting that your code is wrong, which is a known issue IIRC.

**Comment by Vexu** [comment]
Fixed by 5ea79bfc4a0a1269930d98faee36a8f6cb0b0401

```
Semantic Analysis [803/815] ./build/a.zig:28:34: error: expected type '[]i64', found 'std.mem.Error![]i64'
        mem.* = allocator.realloc(mem.*, i + 1);
                                 ^
```
